src/groupByKey.ts

```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items:T[], key: keyof T):GroupsMap<T> {
+  const seed:GroupsMap<T> = {};
```

Comment:
No need for creating a separate variable for this, you could have on line 8
```
items.reduce((accumulator: GroupsMap<T>, item) => {
```
and then on line 18 just pass an empty object `{}`
or you could have instead on line 8 casted the object on line 18 like so
```
 }, {} as GroupsMap<T>); -> i would say this is less preferable
```